hazelcast/src/main/java/com/hazelcast/dataconnection/DataConnectionResource.java

```diff
@@ -88,15 +89,15 @@ public boolean equals(Object o) {
 
         DataConnectionResource dataConnectionResource = (DataConnectionResource) o;
 
-        if (!type.equals(dataConnectionResource.type)) {
+        if (!type.equalsIgnoreCase(dataConnectionResource.type)) {
```

Comment:
Is this the same as 
```suggestion
        if (!type.toLowerCase(Locale.ROOT).equals(dataConnectionResource.type.toLowerCase(Locale.ROOT))) {
```
?

Reply:
it may be not fully equivalent (Unicode...) but `equalsIgnoreCase` should be better: https://youtrack.jetbrains.com/issue/IDEA-217572